Jenkinsfile

```diff
@@ -1,62 +1,39 @@
-		def DOCKER_IMAGE;
-
-        try {
-            stage("Checkout source") {
-                def scmVars = checkout scm
-                GIT_REPOSITORY = scmVars.GIT_URL
-                GIT_BRANCH_NAME = scmVars.GIT_BRANCH
-                GIT_COMMIT_ID = scmVars.GIT_COMMIT
-
-                TIMESTAMP = new java.text.SimpleDateFormat("yyMMddHHmmss").format(new Date())
-            }
-
-            stage("Create version number") {
-                if (env.BRANCH_NAME == 'master') {
-                    ARTIFACT_VERSION = "release-${TIMESTAMP}"
-                } else {
-                    ARTIFACT_VERSION = "pr-${GIT_BRANCH_NAME.replaceAll('/', '_')}-${TIMESTAMP}"
-                }
-            }
-
-            stage("Build opsgenie-heartbeat-proxy image") {
-                DOCKER_IMAGE_NAME = "949398260329.dkr.ecr.eu-west-1.amazonaws.com/opsgenie-heartbeat-proxy:${ARTIFACT_VERSION}";
+properties([
+  disableConcurrentBuilds()
+])
 
-                DOCKER_IMAGE = docker.build(
-                    DOCKER_IMAGE_NAME, 
-                    [
-                        [
-                            "GIT_REPOSITORY=${GIT_REPOSITORY}",
-                            "GIT_BRANCH_NAME=${GIT_BRANCH_NAME}",
-                            "GIT_COMMIT_ID=${GIT_COMMIT_ID}",
-                            "ARTIFACT_VERSION=${ARTIFACT_VERSION}"
-                        ].collect { "--build-arg ${it.toString()}" }.join(" "),
-                        "-f Dockerfile",
-                        "."
-                    ].join(" ")
-                )
-            }
-
-            stage("Publish version") {
-                DOCKER_IMAGE.push(ARTIFACT_VERSION)
-                dockerFingerprintFrom dockerfile: 'Dockerfile', image: DOCKER_IMAGE_NAME
-
-                // jenkins
-                currentBuild.displayName = ARTIFACT_VERSION
-            }
-
-        } finally {
-            stage("Clean directory") {
-                deleteDir();
-            }
+node {
+  ws("workspace/$env.JOB_NAME/$env.BUILD_NUMBER".replace('%2F', '_')) {
+    def SCM_VARS
+    try {
+      stage('Checkout Source') {
+        SCM_VARS = checkout scm
+      }
+
+      stage('Publish Docker Image') {
+        registry = "949398260329.dkr.ecr.eu-west-1.amazonaws.com"
+        fullJobName = env.JOB_NAME.tokenize('/') as String[]
+        timestamp = new java.text.SimpleDateFormat("yyyyMMddHHmmss").format(new Date())
+        projectName = fullJobName[1]
+        version = "$timestamp-${SCM_VARS.GIT_COMMIT[0..7]}"
+
+        if (env.BRANCH_NAME == 'master') {
+          version = "release-$version"
+        } else {
+          version = "pr-${env.BRANCH_NAME.replaceAll('\\W', '_')}-$version"
+        }
+        currentBuild.displayName = version
+        dockerImageName = "$registry/$projectName:$version"
+        dockerImage = docker.build(dockerImageName, '.')
+        dockerImage.push()
+        if (env.BRANCH_NAME == 'master') {
+          DOCKER_IMAGE.push("latest")
```

Comment:
Why `DOCKER_IMAGE` instead of `dockerImage` ?

Reply:
Right, it won't work. It is a typo in our example